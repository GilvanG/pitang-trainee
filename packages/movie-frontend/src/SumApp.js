const Sum = ({ param1 = 0, param2 = 0 }) => {
  return (
    <div>
      <h1>Sum...</h1>

      <span id="sum">{param1 + param2}</span>
    </div>
  );
};

export default Sum;
