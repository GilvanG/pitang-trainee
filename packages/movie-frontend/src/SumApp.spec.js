/* eslint-disable testing-library/no-container */
import { render } from "@testing-library/react";

import SumApp from "./SumApp";

describe("SumApp", () => {
  it("Render SumApp", () => {
    const { asFragment } = render(<SumApp />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("Sum 1 + 1", () => {
    const { container } = render(<SumApp param1={1} param2={2} />);

    const sumSpan = container.querySelector("#sum");

    expect(container.querySelector("h1")).toHaveTextContent("Sum...");

    expect(sumSpan.textContent).toBe("3");
  });
});
