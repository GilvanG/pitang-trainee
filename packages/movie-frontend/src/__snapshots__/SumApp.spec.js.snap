// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SumApp Render SumApp 1`] = `
<DocumentFragment>
  <div>
    <h1>
      Sum...
    </h1>
    <span
      id="sum"
    >
      0
    </span>
  </div>
</DocumentFragment>
`;
